test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpHandler.java

```diff
@@ -374,18 +432,58 @@ public void handle(final HttpExchange exchange) throws IOException {
                 }
             } else {
                 logger.warn("--> Unrecognised request received: {}", request);
-                sendError(exchange, RestStatus.BAD_REQUEST);
+                sendError(exchange, RestStatus.NOT_FOUND, "ResourceNotFound", "The specified resource doesn't exist.");
```

Comment:
Possibly for a follow-up, but (a) could we stick with `BAD_REQUEST` here and (b) I'd much rather we failed the test here, adding more `else if` clauses as necessary to enumerate the specific requests that we expect but do not handle.

Reply:
Done in 952cce2909c15d6ba57d60ea0bc2547a1dd12038

I added specific support for verifying non-existent containers (there was a test for that, I added a read-only version of it for completeness)